jdrfandroidbleparser/src/main/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/compoundcharacteristic/GlucoseMeasurementContextCharacteristic.kt

```diff
@@ -1,13 +1,86 @@
 
+/**
+ * The Glucose Measurement Context characteristic is a variable length structure containing a Flags
+ * field, a Sequence Number field and, based upon the contents of the Flags field, may contain a
+ * Carbohydrate ID field, Carbohydrate field, Meal field, Tester-Health field, Exercise Duration
+ * field, Exercise Intensity field, Medication ID field, Medication field and a HbA1c field.
+ *
+ * https://www.bluetooth.com/specifications/gatt/viewer?attributeXmlFile=org.bluetooth.characteristic.glucose_measurement_context.xml
+ *
+ * @author markiantorno
+ */
 class GlucoseMeasurementContextCharacteristic(characteristic: BluetoothGattCharacteristic?) :
         BaseCharacteristic(characteristic, GattCharacteristic.GLUCOSE_MEASUREMENT_CONTEXT.assigned) {
 
+    override val tag = GlucoseMeasurementContextCharacteristic::class.java.canonicalName as String
+
+    // Sequence Number
+    var sequenceNumber: Int? = null
+    // Field exists if the key of bit 7 of the Flags field is set to 1
+    var extendedFlag: Int? = null
+    // Carbohydrate type, ie: Breakfast, Lunch, etc. Field exists if the key of bit 0 of the Flags field is set to 1
+    var carbohydrateId: CarbohydrateId? = null
+    // Carbohydrate amount in kg. Field exists if the key of bit 0 of the Flags field is set to 1
+    var carbohydrate: Float? = null
+    // Field exists if the key of bit 1 of the Flags field is set to 1
+    var meal: Meal? = null
+    // Field exists if the key of bit 2 of the Flags field is set to 1
+    var tester: Tester? = null
+    // Field exists if the key of bit 2 of the Flags field is set to 1
+    var health: Health? = null
+    // Excercise duration in seconds. Field exists if the key of bit 3 of the Flags field is set to 1.
+    var exerciseDuration: Int? = null
+    // Intensity as percentage. Field exists if the key of bit 3 of the Flags field is set to 1
+    var exerciseIntensity: Float? = null
+    // Insulin type. Field exists if the key of bit 4 of the Flags field is set to 1
+    var medicationId: MedicationId? = null
+    // Unit of measured medication, either kg (mass), or L (volume)
+    var medicationUnit: Units? = null
+    // Numerical measure of medication
+    var medicationAmount: Float? = null
+    // As percentage. Field exists if the key of bit 6 of the Flags field is set to 1
+    var HbA1c: Float? = null
+
     override fun parse(c: BluetoothGattCharacteristic): Boolean {
-        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+        var errorFreeParse = false
+        // These flags define which data fields are present in the Characteristic value
+        Flags.parseFlags(getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT8)).let {
+            sequenceNumber = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT16)
+            if (it.contains(Flags.EXTENDED_FLAGS_PRESENT)) extendedFlag = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT8)
```

Comment:
It is using BluetoothGattCharacteristic.FORMAT_UINT8 instead of org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.FormatType .  Is it intentional?

Reply:
The same for the rest of format type enums

Reply:
Yeah. The local format type is only used internally for calculating offsets. It maps back and fourth under the hood.

Reply:
So to make sure, future reference for FormatType should point to the BluetoothGattCharacteristic version?

Reply:
Any reference to pulling data out of a characteristic uses the offsets defined within that characteristic. I will add more documentation to be ultra specific. But I think I mentioned it in the getNextOffset() method.